### DISCUSSION TITLE
Duplicate snapshot images, and general question about snapshots

### DISCUSSION THREAD

**Discussion by mmcc-xx** [blakeblackshear/frigate#6362] [General]
Hi - I have a camera pointed at my birdfeeder detecting birds, and I wrote some python code that subscribes to frigate/birdcam/bird/snaphot via mqtt and passes the image through another tflite model to determine the species. Overall it works just fine. However, often there are pairs of snapshot events where the same image is sent. I just looked at a few occurrences of this, and in one instance the duplicates were sent 5 seconds apart, and in one it was about 1 minute apart.

Is this expected behavior?

In general, what triggers the sending of a snapshot? Is it when a particular object is detected and then again when it disappears? How is the particular still to send selected?

Thanks,
Clark

**Comment by NickM-27** [comment]
> I just looked at a few occurrences of this, and in one instance the duplicates were sent 5 seconds apart, and in one it was about 1 minute apart.
>
> Is this expected behavior?
>
> In general, what triggers the sending of a snapshot? Is it when a particular object is detected and then again when it disappears? How is the particular still to send selected?

yes it is expected. The mqtt events topioc has `new` when the object is first detected and considered a true positive, `update` which is when a "significant change" has occurred and there can be many updates messages, and finally `end` which is when the object is no longer visible.

a "significant change" can be any number of different things:
- the object enters or exits a zone
- the object has a new snapshot
- the object is stationary for the last minute
- the object is no longer stationary

all of these quality as a "significant change" but that doesn't mean the snapshot will be updated every time. The snapshot update logic depends on the score of the object and if it is better than previously as well as the `best_image_timeout` which is described in further details in the docs

**Reply by mmcc-xx** [reply]
Excellent! Thanks for the information. Is the information about the event (if it is new or an update, e.g.) in the snapshot payload somewhere or should I subscribe to something else as well?

**Reply by NickM-27** [reply]
I was referring to https://docs.frigate.video/integrations/mqtt#frigateevents

**Reply by mmcc-xx** [reply]
Thanks - so, if I wanted just the best scoring image for a particular event, I'd want to monitor frigate/events for events ending for the camera and object type I'm interested in, and then use the http api to retrieve the snapshot for that event. Does that seem like a reasonable approach?

**Reply by NickM-27** [reply]
yes, makes sense to me
